storage/Devices/PartitionImpl.cc

```diff
@@ -481,13 +518,57 @@ namespace storage
     Text
     Partition::Impl::do_set_id_text(Tense tense) const
     {
+	const PartitionTable* partition_table = get_partition_table();
+
 	string tmp = id_to_string(get_id());
 
-	if (tmp.empty())
-	    return sformat(_("Set id of partition %1$s to 0x%2$02X"), get_name().c_str(), get_id());
+	if (partition_table->get_impl().is_partition_id_value_meaningful())
```

Comment:
The intention is to not display e.g. "Set id to BIOS Boot (0x102)" since 0x102 is meaningless to the user. 0x102 is a "random" value in libstorage-ng.

On the other hand displaying e.g. "Set id to EFI System Partition (0xEF)" is meaningful since any user can look the standard 0xEF up.